### PR TITLE
Deprecates avoid_sign_in

### PR DESCRIPTION
**Pull request by kenyonj**
This also refactors the `users_controller_spec` to follow the `expect` syntax.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -1,88 +1,84 @@
...
+require "spec_helper"
 
 describe Clearance::UsersController do
   it { should be_a Clearance::BaseController }
 
-  describe 'when signed out' do
-    before { sign_out }
+  describe "on GET to #new" do
+    context "when signed out" do
+      context "with email" do
```
Comment (derekprior) [inline spec/controllers/users_controller_spec.rb]: I'd get rid of this context and change the order of these two it blocks to say something like:

```
it "renders a form for a new user"
it "renders a form for a user with the query-string-provided email"
```

As a general rule don't use "should" in the it block description.

Comment (derekprior) [inline spec/controllers/users_controller_spec.rb]: Maybe a better description of the email one is:

```
it "defaults email to value provided in query string"
```

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -1,88 +1,84 @@
...
-        @old_user_count = User.count
-        @return_url = '/url_in_the_session'
-        @request.session[:return_to] = @return_url
-        post :create, user: user_attributes
-      end
+  describe "on POST to #create" do
+    context "when signed out" do
+      context "with valid attributes" do
+        it "assigns and creates a user then redirects to the redirect_url" do
+          sign_out
```
Comment (derekprior) [inline spec/controllers/users_controller_spec.rb]: are the `sign_out` calls necessary?

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -1,88 +1,84 @@
...
+          expect(User.count).to eq old_user_count + 1
+          expect(response).to redirect_to(return_url)
+        end
       end
     end
 
-    describe 'POST to create' do
-      before { post :create, user: {} }
+    context "when signed in" do
+      it "redirects to the home page" do
```
Comment (derekprior) [inline spec/controllers/users_controller_spec.rb]: redirects to the configured clearance redirect url.

**Comment by derekprior** [comment]
Few minor comments. Thanks for updating the spec as well.

File: spec/controllers/users_controller_spec.rb (hunk 2 above)
Comment (kenyonj) [inline spec/controllers/users_controller_spec.rb]: no they are not... they were brought over from the previous `before` blocks... I will remove them

**Comment by derekprior** [comment]
LGTM. Assuming travis passes, squash (2 commits looks good here) and merge.

**Pull request merged**